content/blaze.md

```diff
@@ -0,0 +1,559 @@
+
+As an example, consider the `todosItem` template from the Todos example app:
+
+```html
+<template name="todosItem">
+  <div class="list-item {{checkedClass}} {{editingClass}}">
+    <label class="checkbox">
+      <input type="checkbox" checked={{todo.checked}} name="checked">
+      <span class="checkbox-custom"></span>
+    </label>
+
+    <input type="text" value="{{todo.text}}" placeholder="Task name">
+    <a class="js-delete-item delete-item" href="#">
+      <span class="icon-trash"></span>
+    </a>
+  </div>
+</template>
+```
+
+In this example, this template is rendered with an object with key `todo` as data context (we'll see below how to enforce that). We access the properties of the `todo`  using the mustache tag, such as `{{todo.text}}`. The default behaviour is to render that property as a string; however in some cases (such as `checked=={{todo.checked}}` it can be resolved as a boolean value).
+
+Note that simple string interpolations like this will always escape any HTML for you---so you don't need to perform safety checks for XSS.
+
+Additionally we can see an example of a *template helper*---`{{checkedClass}}` calls out to the `checkedClass` helper defined in a separate JavaScript file, which combined with the template defines the `todosItem` component:
+
+```js
+Template.todosItem.helpers({
+  checkedClass() {
+    return this.todo.checked && 'checked';
+  }
+});
+```
+
+In the context of a Blaze helper, `this` is scoped to the current current *data context* at the point the helper was used. This can be hard to reason about, so it's often a good idea to instead pass the required data into the helper as an argument.
+
+Apart from simple interpolation, mustache tags can control flow of the template. For instance, in the `listsShow` template, we render a list of todos via:
+
+```html
+  {{#each todo in todos}}
+    {{> todosItem (todoArgs todo)}}
+  {{else}}
+    <div class="wrapper-message">
+      <div class="title-message">No tasks here</div>
+      <div class="subtitle-message">Add new tasks using the field above</div>
+    </div>
+  {{/each}}
+```
+
+This snippet illustrates a few things:
+
+ - The `{{#each in}}` block helper which renders a block one per element in an array or cursor, or an `{{else}}` block if no items exist.
+ - Template inclusion `{{> todosItem (todoArgs todo)}}` which renders the `todosItem` component with a set data context, based on the output of the `todosArg` helper.
+
+You can read about the full Spacebars syntax [here](https://github.com/meteor/meteor/blob/devel/packages/spacebars/README.md), but in this section we'll attempt to cover some important finer details.
+
+<h3 id="data-contexts">Data contexts and lookup</h3>
+We've seen `{{todo.title}}` accesses the `title` property of the `todo` item on the current data context. Additionaly, `..` access the parent data context (rarely a good idea), `list.todos.[0]` accesses the first element of the `todos` array on `list`.
```

Comment:
Technically, `todo` is not in the data context since it's a lexical variable introduced by `each in`. Not sure how technically correct we need to be.